addons/website_forum/views/website_forum.xml

```diff
@@ -101,6 +96,13 @@
                             </ul>
                         </span>
                     </form>
+                    <ul class="nav navbar-nav pull-right">
```

Comment:
don't rely only on pull-right, it will change position on search box and li in mobile view. define li then search box inside `<ul class="nav navbar-nav navbar-right">`